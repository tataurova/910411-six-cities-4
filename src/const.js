export const PLACES_COUNT = 312;

export const MapSettings = {
  city: [52.38333, 4.9],
  icon: {
    iconUrl: `img/pin.svg`,
    iconSize: [30, 30],
  },
  zoom: 12,
};
