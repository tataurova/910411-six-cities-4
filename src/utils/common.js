export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const mockFunction = () => {
  // do nothing
};
