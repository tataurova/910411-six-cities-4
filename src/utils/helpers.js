export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  for (let key in offers) {
    if (offers.hasOwnProperty(key)) {
      const offer = offers[key].filter((item) => item.id === idNumber)[0];
      if (offer) {
        return {
          offer,
          nearOffers: offers[key],
        };
      }
    }
  }
  return {};
};
