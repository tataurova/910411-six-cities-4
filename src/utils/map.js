import leaflet from "leaflet";

export const addMarkersToMap = (offers, hoveredCardId, icon, activeIcon, map) => {
  const markers = [];
  offers.forEach((offer) => {
    const marker = leaflet.marker(offer.coordinates, {icon: offer.id === hoveredCardId ? activeIcon : icon});
    markers.push(marker);
  });
  const markersGroup = leaflet.layerGroup(markers).addTo(map);
  return markersGroup;
};
