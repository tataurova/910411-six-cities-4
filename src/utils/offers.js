export const getOfferInfo = (offers, offerId) => {
  const idNumber = Number(offerId);
  return {
    offer: offers.filter((item) => item.id === idNumber)[0],
    nearOffers: offers.filter((item) => item.id !== idNumber),
  };
  // for (let key in offers) {
  //   if (offers.hasOwnProperty(key)) {
  //     const offer = offers[key].filter((item) => item.id === idNumber)[0];
  //     if (offer) {
  //       return {
  //         offer,
  //         nearOffers: offers[key].filter((item) => item.id !== offer.id),
  //       };
  //     }
  //   }
  // }
  // return {};
};
