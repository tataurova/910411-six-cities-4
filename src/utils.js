import {SortType} from "./const.js";
import leaflet from "leaflet";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const sortOffers = (sortType, offers) => {
  switch (sortType) {
    case SortType.DEFAULT:
      return offers;
    case SortType.PRICE_UP:
      return offers.sort((a, b) => a.price - b.price);
    case SortType.PRICE_DOWN:
      return offers.sort((a, b) => b.price - a.price);
    case SortType.RATING_DOWN:
      return offers.sort((a, b) => b.rating - a.rating);
  }
  return offers;
};

export const getSortTypeName = (sortType) => {
  switch (sortType) {
    case SortType.PRICE_UP:
      return `Price: low to high`;
    case SortType.PRICE_DOWN:
      return `Price: high to low`;
    case SortType.RATING_DOWN:
      return `Top rated first`;
  }
  return `Popular`;
};

export const getArrayOfOffers = (citiesOffers) => {
  let allOffers = [];
  Object.values(citiesOffers).forEach((item) => {
    allOffers = allOffers.concat(item);
  });
  return allOffers;
};

export const addMarkersToMap = (offers, hoveredCardId, icon, activeIcon, map) => {
  const markers = [];
  offers.forEach((offer) => {
    const marker = leaflet.marker(offer.coordinates, {icon: offer.id === hoveredCardId ? activeIcon : icon});
    markers.push(marker);
  });
  const markersGroup = leaflet.layerGroup(markers).addTo(map);
  return markersGroup;
};

export const getNearOffers = (offers, id, city) => {
  const numberId = Number(id);
  return offers.filter((offer) => offer.id !== numberId && offer.city === city);
};

export const findOfferById = (offers, id) => {
  return offers.filter((offer) => offer.id === id)[0];
};
