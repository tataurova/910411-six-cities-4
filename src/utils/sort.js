import {SortType} from "../const.js";

export const sortOffers = (sortType, offers) => {
  switch (sortType) {
    case SortType.DEFAULT:
      return offers;
    case SortType.PRICE_UP:
      return offers.sort((a, b) => a.price - b.price);
    case SortType.PRICE_DOWN:
      return offers.sort((a, b) => b.price - a.price);
    case SortType.RATING_DOWN:
      return offers.sort((a, b) => b.rating - a.rating);
  }
  return offers;
};

export const getSortTypeName = (sortType) => {
  switch (sortType) {
    case SortType.PRICE_UP:
      return `Price: low to high`;
    case SortType.PRICE_DOWN:
      return `Price: high to low`;
    case SortType.RATING_DOWN:
      return `Top rated first`;
  }
  return `Popular`;
};
