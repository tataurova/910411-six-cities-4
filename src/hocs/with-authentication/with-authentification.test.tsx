import * as React from "react";
import * as renderer from "react-test-renderer";
import withAuthentication from "./with-authentication";
import {mockFunction} from "../../utils/common";
import {AuthorizationStatus} from "../../const";

const MockComponent = () => {
  return (
    <div></div>
  );
};

const MockComponentWrapped = withAuthentication(MockComponent);

it(`withAuthentication is rendered correctly`, () => {
  const onSubmitForm = (mockFunction);
  const tree = renderer.create((
    <MockComponentWrapped
      onSubmitForm = {onSubmitForm}
      authorizationStatus = {AuthorizationStatus.NO_AUTH}
      user = {``}
      error = {false}
    />
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
