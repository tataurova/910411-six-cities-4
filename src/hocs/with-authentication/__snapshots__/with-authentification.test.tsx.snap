// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuthentication is rendered correctly 1`] = `<div />`;
